packages/integrations/image/components/Image.astro

```diff
@@ -0,0 +1,100 @@
+---
+// @ts-ignore
+import loader from 'virtual:image-loader';
+import { getImage } from '../src';
+import type { ImageAttributes, ImageMetadata, TransformOptions, OutputFormat } from '../src/types';
+
+export interface LocalImageProps extends Omit<TransformOptions, 'src'>, Omit<ImageAttributes, 'src'> {
+	src: ImageMetadata;	
+}
+
+export interface RemoteImageProps extends TransformOptions, ImageAttributes {
+	src: string;
+	format: OutputFormat;
+	width: number;
+	height: number;
+}
+
+export type Props = LocalImageProps | RemoteImageProps;
+
+function isLocalImage(props: Props): props is LocalImageProps {
+	// vite-plugin-astro-image resolves ESM imported images
+	// to a metadata object
+	return typeof props.src !== 'string';
+}
+
+function resolveProps(props: Props): TransformOptions {
+	if (!isLocalImage(props)) {
+		return props;
+	}
+
+	let { width, height, aspectRatio, format, ...rest } = props;
+	const { src, ...metadata } = props.src;
+
+	if (!width && !height) {
+		width = metadata.width;
+		height = metadata.height;
+	} else if (height || width) {
+		aspectRatio = aspectRatio || metadata.width / metadata.height;
+	}
+	
+	return {
+		...rest,
+		width,
+		height,
+		aspectRatio,
+		src,
+		format: format || metadata.format as OutputFormat,
+	}
+}
+
+function calculateSize(transform: TransformOptions): TransformOptions {
+	// keep width & height as provided
+	if (transform.width && transform.height) {
+		return transform;
+	}
+
+	if (!transform.width && !transform.height) {
+		throw new Error(`"width" and "height" cannot both be undefined`);
+	}
```

Comment:
I see that the `metadata` function is being used to calculate the size in the Vite plugin, what are the situations where we need an explicit size passed in?

Reply:
Users need to provide dimensions for remote images.  The Vite plugin handles any local ESM imports and the width/height from there is used for local images